src/Hastlayer/Hast.Transformer/Services/ConstantValuesSubstitution/ConstantValuesSubstitutingVisitor.cs

```diff
@@ -421,7 +421,7 @@ public override void VisitAssignmentExpression(AssignmentExpression assignmentEx
             {
                 ConstantValuesTable.MarkAsNonConstant(assignmentExpression.Left, _constructor);
             }
-            else if (right != null)
```

Comment:
Why can this be removed?

Reply:
If `right` was null then the `assignmentExpression.Right is not PrimitiveExpression right` in the branch above would've triggered. So this condition is always true.